src/integrators/interface.jl

```diff
@@ -221,8 +221,28 @@ end
 
 DiffEqBase.change_t_via_interpolation!(integrator::SDDEIntegrator,t,modify_save_endpoint::Type{Val{T}}=Val{false}) where T = StochasticDiffEq.change_t_via_interpolation!(integrator, t, modify_save_endpoint)
 
-function u_modified!(integrator::SDDEIntegrator,bool::Bool)
-  integrator.u_modified = bool
+# update integrator when u is modified by callbacks
+function StochasticDiffEq.handle_callback_modifiers!(integrator::SDDEIntegrator)
+    # copied from StochasticDiffEq
+    if integrator.P !== nothing && integrator.opts.adaptive
+        if integrator.cache isa StochasticDiffEqMutableCache
+            oldrate = integrator.P.cache.currate
+            integrator.P.cache.rate(oldrate, integrator.u, integrator.p, integrator.t)
+        else
+            integrator.P.cache.currate = integrator.P.cache.rate(
+                integrator.u, integrator.p, integrator.t
+            )
+        end
+    end
```

Comment:
The original code in StochasticDiffEq is incorrect (unqualified `u`, `p`, `t`, and `P`), it seems it is not tested properly.